chore: react-combobox spec and migration docs update (#26396)
microsoft/fluentui: packages/react-components/react-combobox/docs/Spec.md

## Background

The basic purpose of a Combobox or Dropdown is to allow users to select one or more values from among a set of options. The semantics and behavior are roughly similar to a more complex version of an HTML `<select>` element, with more functionality and styling control.

A combobox can be single or multi-select, and it can be editable (`<Combobox>`) or select-only (`<Dropdown>`). More options are covered in the [Variants section](#variants).

The basic structure of a Combobox or Dropdown has two pieces: the faceplate, which is always rendered on the page, and displays the current selection, and the popup that contains a list of options.

### Combobox vs. Dropdown

> 

Reviewer: Looks like this was cut off.

Author: Listbox isn't provided as a standalone widget anymore, so I removed that section from the docs

Author: oh, haha, didn't realize the sentence below was cut off 😂

The `react-combobox` package provides two combobox-like selection controls: `<Combobox>` and `<Dropdown>`. While they share the bulk of their logic under the hood, they differ in the

> ### Combobox/Dropdown vs. Select vs. Menu

Reviewer: ```suggestion
### Combobox vs. Dropdown vs. Select vs. Menu
```

> Use Select when a basic single-select form component with no freeform text input or filtering is needed.

Reviewer: It's a restatement of the previous paragraph but adding something like:

```suggestion
Use Select when a basic single-select form component with no freeform text input or filtering is needed; or when you need the best possible mobile support.
```

> ### When to use Combobox or Dropdown

Reviewer: Consider breaking this into two sections, one for Combobox and one for Dropdown. Above you state that the controls discussed in this section cannot be used interchangeably but then these two controls are grouped together implying interchangeability.

Author: I was aiming to break them up by package, but I definitely see how it was confusing. I've updated the headings and wording a bit -- let me know if that reads OK, or if you think I should still just merge the two sections together.

> Dropdown is a more feature-rich version of Select, which comes at the cost a larger code footprint, and less robust support for accessibility compared to the native `<select>` element. Combobox (as mentioned previously) is essentially a Dropdown that allows text input.

Reviewer: I'd drop the "(as mentioned previously)" paranthetical. Documentation isn't always read top to bottom so it's reasonable that a read won't have seen this previous mention.

Use Dropdown over Select when any of the following are required:

- Virtualization
- Control over styling the dropdown and options
- Multiple selection

> Use Combobox when filtering or freeform text input is needed.

Reviewer: ```suggestion
Use Combobox when you need any of Dropdown's features plus 
- filtering 
- freeform text input
```

Author: Tweaked the wording slightly (actually had already changed it based on one of your earlier comments 😅). Let me know if that looks good!
